### PR TITLE
[BP] - create json exporter for report

### PR DESCRIPTION
**Pull request by c4bae**
## Summary by Sourcery

Add JSON export support for shelter occupancy reporting metrics and corresponding validation tests.

New Features:
- Introduce a JSON exporter that serializes selected shelter occupancy metrics into a date-keyed report file.

Tests:
- Add tests covering JSON metric export structure, option filtering, and validation for empty or unknown export options.

### CODE REVIEW

File: apps/betterangels-backend/reports/export_to_json.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    shelter_id = str(metrics.shelter_id)
+    start_date = metrics.start_date
+    end_date = metrics.end_date
+    date_range = f"{start_date.isoformat()}_{end_date.isoformat()}"
+    report: dict[str, object] = {}
+
+    if MetricsExportOptions.DAILY_OCCUPANCY_METRICS in selected_options:
+        report[MetricsExportOptions.DAILY_OCCUPANCY_METRICS] = {
+            metric.date.isoformat(): {
+                "shelter_id": shelter_id,
```
Comment (frank731) [inline apps/betterangels-backend/reports/export_to_json.py]: does shelter id need to be in every exported metric, in every day, could it just be at the top level

**Review by frank731 — COMMENTED** [review]
One nit otherwise should be solid

File: apps/betterangels-backend/reports/tests/test_exporters.py [hunk 2]
```diff
@@ -402,18 +402,16 @@ def test_metrics_to_json_exports_selected_metrics_as_date_keyed_report(self) ->
 
         assert filename == "20260601_20260630_shelter_report.json"
         assert json.loads(json_content) == {
-            "report": {
+            "shelter-1_report": {
```
Comment (frank731) [inline apps/betterangels-backend/reports/tests/test_exporters.py]: this feels like it would cleaner if shelter id was still a field under report, feel like it would be hard to ingest if the schema is diff for each report

**Review by frank731 — CHANGES_REQUESTED** [review]

**Review by frank731 — APPROVED** [review]
